### PR TITLE
docs: Explain the dependencies for the output of the Java codegen.

### PR DESCRIPTION
**Pull request by gerolf-da**
### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](./CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](./docs/source/support/release-notes.rst), if appropriate

### CODE REVIEW

File: docs/source/app-dev/bindings-java/codegen.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ When writing applications for the ledger in Java, you want to work with a repres
 Download
 ========
 
-You can download the `latest version <https://bintray.com/api/v1/content/digitalassetsdk/DigitalAssetSDK/com/daml/java/codegen/$latest/codegen-$latest.jar?bt_package=sdk-components>`__  of the Java codegen.
+You can download the `latest version <https://bintray.com/api/v1/content/digitalassetsdk/DigitalAssetSDK/com/daml/java/codegen/$latest/codegen-$latest.jar?bt_package=sdk-components>`__  of the Java codegen. Make sure the Java codegen version is the same as the SDK version you are using (see :ref:`assistant-manual-managing-releases`).
```
Comment (bethaitman-da) [inline docs/source/app-dev/bindings-java/codegen.rst]: "Make sure the Java codegen version is the same as the SDK version" How do I actually check that in practise - ie, where do I look?

File: docs/source/app-dev/bindings-java/codegen.rst [hunk 2]
```diff
@@ -92,6 +92,22 @@ The following snippet is an excerpt from the ``pom.xml`` that is part of the :re
     :lines: 73-105,121-122
     :dedent: 12
 
+
+Compile the generated Java code
+===============================
+
+To compile the generated Java code, add the :ref:`Java Bindings <bindings-java-setup-maven>` library with the same version as the Java codegen to the classpath. With Maven you can do this by adding a ``dependency`` to the ``pom.xml`` file:
```
Comment (bethaitman-da) [inline docs/source/app-dev/bindings-java/codegen.rst]:
```suggestion
To compile the generated Java code, add the :ref:`Java Bindings <bindings-java-setup-maven>` library with the same version as the Java codegen to the classpath. 

With Maven you can do this by adding a ``dependency`` to the ``pom.xml`` file:
```

File: docs/source/app-dev/bindings-java/codegen.rst (hunk 1 above)
Comment (gerolf-da) [reply docs/source/app-dev/bindings-java/codegen.rst]: So the combo to check is:
- downloaded java codegen jar: actively done by the user
- maven dependency in pom.xml: comes initially from the quickstart, which depends on the SDK version
- SDK version

Comment (gerolf-da) [reply docs/source/app-dev/bindings-java/codegen.rst]: How about something like:

```
Make sure that the following SDK component versions are the same:
* the downloaded Java codegen jar file, eg. 10x.y.z
* the dependency to bindings-java, eg. 10x.y.z
* the sdk-version attribute in the da.yaml file, eg. x.y.z
```

Comment (bethaitman-da) [reply docs/source/app-dev/bindings-java/codegen.rst]: That looks perfect!

**Comment by gerolf-da** [comment]
Thanks @bethaitman-da: I now also link to to the da.yaml configuration section in the assistant docs.

**Pull request merged**